File: tests/fixtures/adapters/ts_js/sample.ts
export interface Runner {
  run(input: string): string;
}

export enum Mode {
  Fast,
  Slow,
}

export type Result = {
  ok: boolean;
};

export class Service {
  constructor(private readonly name: string) {}

  async run(input: string): Promise<string> {
    return `${this.name}:${input}`;
  }

  format(value: string): string {
    return value.trim();
  }
}

export async function build(name: string): Promise<Service> {
  return new Service(name);
}

export const DEFAULT_NAME = "service";

File: tests/fixtures/adapters/go/sample.go
package worker

import "context"

type Runner interface {
	Run(ctx context.Context) error
}

type Service struct {
	name string
}

const (
	DefaultName = "svc"
	MaxRetries  = 3
)

var (
	GlobalEnabled = true
	globalVersion = "dev"
)

func Build(name string) *Service {
	return &Service{name: name}
}

func (s *Service) Run(ctx context.Context) error {
	_ = ctx
	return nil
}
